src/lib.rs

```diff
@@ -162,6 +162,10 @@ pub enum Level {
     Best,
     /// Default quality of compression defined by the selected compression algorithm.
     Default,
+    /// Precise quality based on the underlying compression algorithms'
+    /// qualities. The interpretation of this depends on the algorithm choosen.
+    /// For bzip2, this is treated as default.
```

Comment:
I think the special case for bzip2 doesn't really need to be mentioned, it's implicit in this value being implementation specific anyway.

I also think that the fact that this is _implementation_ specific, not just _algorithm_ specific should be highlighted. The main reason I want this local enum instead of just re-exporting the underlying configuration from the different implementations is to allow switching implementations of algorithms in the future (e.g. I would really love to drop all the C dependencies at some point when viable Rust replacements are available).

```suggestion
    /// qualities. The interpretation of this depends on the algorithm chosen
    /// and the specific implementation backing it.
```